sycl/unittests/scheduler/LeafLimit.cpp

```diff
@@ -18,40 +18,43 @@ using namespace cl::sycl;
 
 TEST_F(SchedulerTest, LeafLimit) {
   MockScheduler MS;
+  std::vector<std::unique_ptr<MockCommand>> LeavesToAdd;
+  std::unique_ptr<MockCommand> MockDepCmd;
 
   buffer<int, 1> Buf(range<1>(1));
   detail::Requirement MockReq = getMockRequirement(Buf);
-  MockCommand *MockDepCmd =
-      new MockCommand(detail::getSyclObjImpl(MQueue), MockReq);
+
+  MockDepCmd =
+      std::make_unique<MockCommand>(detail::getSyclObjImpl(MQueue), MockReq);
   detail::MemObjRecord *Rec =
       MS.getOrInsertMemObjRecord(detail::getSyclObjImpl(MQueue), &MockReq);
```

Comment:
Could you please commit these refactoring as a separate PR?

Reply:
Those changes fix memory leaks in the `LeafLimit` test. They are exposed (as `Gtest` warnings) as a result of my changes to `MockCommand`. I initially planned to fix memory leaks later, but testing revealed new failures.

Reply:
Ok then.